src/f32-avgpool/gen/f32-avgpool-9x-minmax-rvv-c1v.c

```diff
@@ -27,96 +27,37 @@ void xnn_f32_avgpool_minmax_ukernel_9x__rvv_c1v(
   assert(kernel_elements != 0);
   assert(kernel_elements <= 9);
   assert(channels != 0);
+  assert((input_offset & 3) == 0);
+
+  input_offset >>= XNN_LOG2_SIZEOF_FLOAT;
 
   const float scale = params->scalar.scale;
   const float min = params->scalar.min;
   const float max = params->scalar.max;
 
   do {
-    const float* i0 = input[0];
-    assert(i0 != NULL);
-    const float* i1 = input[1];
-    const float* i2 = input[2];
-    const float* i3 = input[3];
-    const float* i4 = input[4];
-    const float* i5 = input[5];
-    const float* i6 = input[6];
-    const float* i7 = input[7];
-    const float* i8 = input[8];
-    input = (const float**) ((uintptr_t) input + input_increment);
-    if (kernel_elements < 2) {
-      i1 = zero;
-    }
-    assert(i1 != NULL);
-    if (kernel_elements <= 2) {
-      i2 = zero;
-    }
-    assert(i2 != NULL);
-    if (kernel_elements < 4) {
-      i3 = zero;
-    }
-    assert(i3 != NULL);
-    if (kernel_elements <= 4) {
-      i4 = zero;
-    }
-    assert(i4 != NULL);
-    if (kernel_elements < 6) {
-      i5 = zero;
-    }
-    assert(i5 != NULL);
-    if (kernel_elements <= 6) {
-      i6 = zero;
-    }
-    assert(i6 != NULL);
-    if (kernel_elements < 8) {
-      i7 = zero;
-    }
-    assert(i7 != NULL);
-    if (kernel_elements <= 8) {
-      i8 = zero;
-    }
-    assert(i8 != NULL);
-    if XNN_UNPREDICTABLE(i0 != zero) {
-      i0 = (const float*) ((uintptr_t) i0 + input_offset);
+    const float *i[9];
+    for (size_t kk = 0; kk < kernel_elements; ++kk) {
```

Comment:
I think that `kernel_elements` can be greater than 9 here, so these bounds need rethinking

Reply:
No. In this case there is assert(kernel_elements <= 9); at start.